Modules/DialogManagement/src/Game.DialogManagement.Domain/Data/ValueObjects/DialogueContext.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using Game.Shared.Core.Data;
+
+namespace Game.DialogManagement.Domain.Data.ValueObjects
+{
+    /// <summary>
+    /// The dialogue context, containing all of the information neccessary to
+    /// actually converse.
+    /// </summary>
+    /// <param name="NpcName">The name of the NPC we're talking with.</param>
+    /// <param name="NpcStory">The backstory of said NPC.</param>
+    /// <param name="PlayerName">The name of the player talking.</param>
+    /// <param name="Backlog">The backlog of messages.</param>
+    public record DialogueContext(
```

Comment:
_DialogueContext_ jest tak jak rozmawialiśmy na spotkaniu ValueObject'em i może tak zostać, bo jest poprawnie, tylko informacyjnie valueobjecty zazwyczaj są mniejsze, że to np. NpcName jest ValueObjectem oraz NpcStory, a taki DialogueContext  jako entity uważam, że tutaj również pasuje, szczególnie, że dialoguecontext, a dokładniej Backlog w nim ma jakąś ciągłość i prędzej string'i Backlog'u potraktowałbym jako valueobject i nawet zalecam tak zrobić z jednym polem np. Text, na wypadek jakbyśmy kiedyś chcieli coś dodać np. poziom nastawienia wiadomości(przyjazna/wroga), lecz DialogueContext jako valueobject w naszym systemie też pasuje i może tak zostać.

Reply:
tutaj akurat po głębszym przemyśleniu nie wiem czy DialogueContext powinien zostać jako ValueObject, ponieważ niby powinien być mutowany (dodajemy kolejne linijki dialogu) po każdym wywołaniu metody Chat w IAiApiProvider...

to akurat zostawię do twojej decyzji

Reply:
Tak możesz wydzielić jako entitiy i tak samo NPC bym wydzielił jako entity, bo on z czasem może mieć bardzo dużo pól i zachowań. ValueObjectem na pewno będą elementy kolekcji BackLog'u